### PR TITLE
Cleanup unused error pages

### PR DESCRIPTION
**Pull request by pablothedude**
Unused or misleading exceptions are changed to become more clear or removed if dead.

See:
https://www.pivotaltracker.com/story/show/167452099
https://www.pivotaltracker.com/story/show/167201727

### CODE REVIEW

File: languages/messages.nl.php [hunk 1]
```diff
@@ -187,14 +187,14 @@
     'error_unsupported_signature_method_desc' => 'De ondertekeningsmethode %arg1% wordt niet ondersteund, upgrade naar RSA-SHA256 (http://www.w3.org/2001/04/xmldsig-more#rsa-sha256).',
     'error_unknown_preselected_idp' => 'Fout - Dienst niet toegankelijk via %organisationNoun%',
     'error_unknown_preselected_idp_desc' => 'De %organisationNoun% waarmee je wilt inloggen heeft toegang tot deze dienst niet geactiveerd. Dat betekent dat jij geen gebruik kunt maken van deze dienst via %suiteName%. Neem contact op met de helpdesk van jouw %organisationNoun% als je toegang wilt krijgen tot deze dienst. Geef daarbij aan om welke dienst het gaat (de &lsquo;SP&rsquo;) en waarom je toegang wilt.',
-    'error_unknown_service_provider'              => 'Error - Unkown service',
+    'error_unknown_service_provider'              => 'Error - Onbekende service',
```
Comment (MKodde) [inline languages/messages.nl.php:190]: I think this translation is better translated using 'dienst' instead of 'service'. Reading other SP related error messages, the SP is usually called a 'dienst'

File: languages/messages.pt.php [hunk 2]
```diff
@@ -239,8 +239,8 @@
     'error_attribute_validator_not_in_definitions'  => '%arg1% não é conhecido no schema',
     'error_attribute_validator_allowed'             => '\'%arg3%\' não é um valor permitido para este atributo',
     'error_attribute_validator_availability'        => '\'%arg3%\' é um schacHomeOrganization reservado para outro Fornecedor de Identidade',
-    'error_unknown_service'         => 'Erro - Serviço desconhecido',
-    'error_unknown_service_desc'    => '<p>O serviço solicitado não foi encontrado.</p>',
+    'error_unknown_requesterid_in_authnrequest'         => 'Erro - Serviço desconhecido',
+    'error_unknown_requesterid_in_authnrequest_desc'    => '<p>O serviço solicitado não foi encontrado.</p>',
```
Comment (MKodde) [inline languages/messages.pt.php:243]: Maybe ask Dominguez to co-read the Portuguese changes?

File: library/EngineBlock/Exception/UnknownIdentityProvider.php [hunk 3]
```diff
@@ -16,14 +16,16 @@
  * limitations under the License.
  */
 
-class EngineBlock_Corto_Exception_UnknownIssuer extends EngineBlock_Exception
+use OpenConext\EngineBlock\Metadata\Entity\ServiceProvider;
+
+class EngineBlock_Exception_UnknownIdentityProvider extends \Exception
```
Comment (MKodde) [inline library/EngineBlock/Exception/UnknownIdentityProvider.php:21]: Why did you change the extension to `\Exception` instead of `EngineBlock_Extension`? Also if you keep the 'base' \Exception, please import it using a `use` statement.

File: src/OpenConext/EngineBlockFunctionalTestingBundle/Features/ClearErrorMessages.feature [hunk 4]
```diff
@@ -137,6 +138,20 @@ Feature:
       And I should see "IdP:"
       And I should see "https://wrong.example.edu/metadata"
 
+  Scenario: A Trusted proxy with which requireds a RequesterId and when one is omitted
```
Comment (MKodde) [inline src/OpenConext/EngineBlockFunctionalTestingBundle/Features/ClearErrorMessages.feature:141]: Typo/suggestion `A Trusted proxy with which requireds a RequesterId and when one is omitted`

A trusted proxy which requires a RequesterID but did not send it is not recognized by EB

File: src/OpenConext/EngineBlockFunctionalTestingBundle/Features/SpProxy.feature [hunk 5]
```diff
@@ -222,14 +222,14 @@ Feature:
       And SP "Step Up" does not require consent
       And SP "Step Up" uses the Unspecified NameID format
     When I log in at "Step Up"
-    Then I should see "Error - Unknown service"
+    Then I should see "Unknown service"
```
Comment (MKodde) [inline src/OpenConext/EngineBlockFunctionalTestingBundle/Features/SpProxy.feature:225]: Was removing the `Error - ` part also required. I did not read about it in the story.

File: languages/messages.en.php [hunk 6]
```diff
@@ -189,13 +189,13 @@
...
     'error_unknown_preselected_idp_desc' => 'The %organisationNoun% that you want to use to login to this service did not activate access to this service. This means you are unable to use this service through %suiteName%. Please contact the helpdesk of your %organisationNoun% to request access to this service. State what service it is about (the &lsquo;SP&rsquo;) and why you need access.',
-    'error_unknown_service_provider'          => 'Error - Unkown service',
+    'error_unknown_service_provider'          => 'Error - Unknown service',
     'error_unknown_service_provider_desc'     => 'No Service Provider could be found with the EntityID \'%arg1%\'.',
 
     'error_unsupported_acs_location_scheme' => 'Error - Unsupported URI scheme in ACS location',
 
-    'error_unknown_issuer'          => 'Error - Unknown service',
-    'error_unknown_issuer_desc'     => 'The service you are trying to log in to is unknown to %suiteName%. Possibly your %organisationNoun% has never enabled access to this service. If you would like to use this service, please contact the helpdesk of your %organisationNoun%.',
```
Comment (MKodde) [inline languages/messages.en.php:205]: PM Thijs suggested the following: `Copy texts from unknown_issuer template to unknown_service_provider.`

This does not seem to have happened. Was this intentional?

File: languages/messages.nl.php (hunk 1 above)
Comment (pablothedude) [reply languages/messages.nl.php:190]: Was fixed by copying the `unknown_issuer` texts:
`https://github.com/OpenConext/OpenConext-engineblock/pull/849#discussion_r422851429`

File: languages/messages.pt.php (hunk 2 above)
Comment (pablothedude) [reply languages/messages.pt.php:243]: I don't want that to be blocking for this PR. I've notified Domgon that the languages are available when 6.2 is released. So when 6.2 lands I'll notify him again.
`https://github.com/OpenConext/OpenConext-engineblock/pull/650#issuecomment-618945881`

File: library/EngineBlock/Exception/UnknownIdentityProvider.php (hunk 3 above)
Comment (pablothedude) [reply library/EngineBlock/Exception/UnknownIdentityProvider.php:21]: Because I renamed the UnknownIssuer exception. It had to be replaced by the UnknownServiceProvider exception which had a different footprint, so I changed that accordingly.

File: src/OpenConext/EngineBlockFunctionalTestingBundle/Features/ClearErrorMessages.feature (hunk 4 above)
Comment (pablothedude) [reply src/OpenConext/EngineBlockFunctionalTestingBundle/Features/ClearErrorMessages.feature:141]: Changed

File: src/OpenConext/EngineBlockFunctionalTestingBundle/Features/SpProxy.feature (hunk 5 above)
Comment (pablothedude) [reply src/OpenConext/EngineBlockFunctionalTestingBundle/Features/SpProxy.feature:225]: No this wasn't required, but I don't think all changes should be mentioned in stories either (boyscouting for example). I think you could even argue why a SP Proxy functional test needs to have `Error - ` in this test? But that's a different discussion.

I've used it to replicate an exception which was added without test coverage and didn't revert it properly in this PR. I think we also need to be careful when contributions are made that tests are also added.
`https://github.com/OpenConext/OpenConext-engineblock/pull/540`

I however will revert this change to keep the commit more atomic.

File: languages/messages.en.php (hunk 6 above)
Comment (pablothedude) [reply languages/messages.en.php:205]: Good catch! done!

File: library/EngineBlock/Exception/UnknownRequesterIdInAuthnRequest.php [hunk 7]
```diff
@@ -16,6 +16,21 @@
  * limitations under the License.
  */
 
-class EngineBlock_Corto_Module_Bindings_UnknownIssuerException extends EngineBlock_Corto_Module_Bindings_VerificationException
+use OpenConext\EngineBlock\Metadata\Entity\ServiceProvider;
+
+class EngineBlock_Exception_UnknownRequesterIdInAuthnRequest extends \Exception
```
Comment (MKodde) [inline library/EngineBlock/Exception/UnknownRequesterIdInAuthnRequest.php:21]: Nitpick: for the sake of consistency you could import this \Exception

File: languages/messages.nl.php (hunk 1 above)
Comment (MKodde) [reply languages/messages.nl.php:190]: I still see `Onbekende service`, or do I misinterpret your response? 

FYI The old value for the unknow issuer text also spoke of 'dienst'.

File: library/EngineBlock/Exception/UnknownRequesterIdInAuthnRequest.php (hunk 7 above)
Comment (pablothedude) [reply library/EngineBlock/Exception/UnknownRequesterIdInAuthnRequest.php:21]: I tried, but that resulted in an Exception:
`https://travis-ci.org/github/OpenConext/OpenConext-engineblock/jobs/685589429#L1029`
